### PR TITLE
Finish flattening /prim tests

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This is the first of 2 PRs to flatten /prim. Flattening the rest of /prim in one PR is not feasible for reviewing. This PR flattens the tests and creates a simple master header `<stan/math/prim.hpp>` in order to replace the prim/arr.hpp, prim/mat.hpp and prim/scal.hpp in the tests.

The PR touches A LOT of files, but the changes are simple. Only a few files had to be merged.

## Checklist

- [x] Math issue #937 

- [x] Copyright holder: Rok Češnovar

### CODE REVIEW

File: test/unit/math/prim/util.hpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef TEST_MATH_UNIT_PRIM_MAT_UTIL_HPP
 #define TEST_MATH_UNIT_PRIM_MAT_UTIL_HPP
```
Comment (mcol) [inline test/unit/math/prim/util.hpp]: Header guard.

File: test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp [hunk 2]
```diff
@@ -1,10 +1,6 @@
 #include <stan/math/rev.hpp>
-#include <stan/math.hpp>
+#include <stan/math/prim.hpp>
 #include <gtest/gtest.h>
-#include <stan/math/prim/scal/fun/value_of.hpp>
-#include <stan/math/prim/meta/return_type.hpp>
```
Comment (mcol) [inline test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp]: Shouldn't we keep `stan/math/prim/meta.hpp`?

File: test/unit/math/prim/fun/promote_elements_test.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  EXPECT_TRUE(same);
+}
+
+TEST(MathFunctionsScalPromote_Elements, double2var) {
+  double from;
+  promote_elements<var, double> p;
+  typedef BOOST_TYPEOF(p.promote(from)) result_t;
+  bool same = std::is_same<var, result_t>::value;
+  EXPECT_TRUE(same);
+}
```
[the remark this answers is not in the document]
Comment (rok-cesnovar) [reply test/unit/math/prim/fun/promote_elements_test.cpp]: Good call yeah, made test/unit/math/rev/fun/promote_elements_test.cpp and moved all tests that include var there.

File: test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp (hunk 2 above)
Comment (rok-cesnovar) [reply test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp]: This includes the prim master header, so I thought it doesnt make sense to also include meta.

**Comment by rok-cesnovar** [comment]
Thanks @mcol! Only a few of them left :) Will probably split the rest in two more, to not have these 700 file PRs :)

File: test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp (hunk 2 above)
Comment (mcol) [reply test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp]: OK, then note that `test/unit/math/rev/meta/operands_and_partials_test.cpp` includes both.

Comment (rok-cesnovar) [reply test/unit/math/rev/prob/categorical_logit_glm_lpmf_test.cpp]: Removed that. Thanks!

**Pull request merged**